bpf/lib/dbg.h

```diff
@@ -236,11 +236,65 @@ static __always_inline void cilium_dbg_capture2(struct __ctx_buff *ctx, __u8 typ
 			 &msg, sizeof(msg));
 }
 
-static __always_inline void cilium_dbg_capture(struct __ctx_buff *ctx, __u8 type,
-					       __u32 arg1)
+#elif defined(DEBUG_TAGGED)
+#include "trace_helpers.h"
+
+# define printk(fmt, ...)					\
+		do { } while (0)
+
+static __always_inline void cilium_dbg(struct __ctx_buff *ctx, __u8 type,
```

Comment:
IMO it would be better to modify `cilium_dbg` and co above directly instead of reimplementing them here. It would also make the diff smaller and clearer. Does this work?:

```c
static __always_inline void cilium_dbg(struct __ctx_buff *ctx, __u8 type,
				       __u32 arg1, __u32 arg2)
{
	struct debug_msg msg = {
		__notify_common_hdr(CILIUM_NOTIFY_DBG_MSG, type),
		.arg1	= arg1,
		.arg2	= arg2,
	};
#ifdef DEBUG_TAGGED
    if (!load_ip_trace_id())
		return;
#endif
	ctx_event_output(ctx, &cilium_events, BPF_F_CURRENT_CPU,
			 &msg, sizeof(msg));
}
```